File: zigar-runtime/test/doc-examples/print-struct.zig
// print-struct.zig
const std = @import("std");

pub const StructA = struct {
    number1: i16,
    number2: i16,
};

pub const StructB = struct {
    a: ?StructA,
    number3: f64,
};

pub fn printStruct(s: StructB) void {
    std.debug.print("{any}\n", .{s});
}

File: zigar-runtime/test/doc-examples/print-struct.js
// print-struct.js
import { printStruct } from './print-struct.zig';

printStruct({ a: { number1: 123, number2: 456 }, number3: 77 });
printStruct({ a: null, number3: 77 });

// console output:
// print-struct.StructB{ .a = print-struct.StructA{ .number1 = 123, .number2 = 456 }, .number3 = 7.7e+01 }
// print-struct.StructB{ .a = null, .number3 = 7.7e+01 }
